ui/frontend/BuildMenu.tsx

```diff
@@ -24,29 +26,44 @@ interface BuildMenuProps {
   close: () => void;
 }
 
-const BuildMenu: React.SFC<BuildMenuProps> = props => (
-  <MenuGroup title="What do you want to do?">
-    <ButtonMenuItem name="Build" onClick={() => { props.execute(); props.close(); }}>
-      No bells and whistles, regular build coming right&nbsp;up&nbsp;:D
-    </ButtonMenuItem>
-    <ButtonMenuItem name="ASM" onClick={() => { props.compileToAssembly(); props.close(); }}>
-      Build and show the resulting assembly code.
-    </ButtonMenuItem>
-    <ButtonMenuItem name="LLVM IR" onClick={() => { props.compileToLLVM(); props.close(); }}>
-      Build and show the resulting LLVM IR, LLVM's intermediate representation.
-    </ButtonMenuItem>
-    <ButtonMenuItem name="MIR" onClick={() => { props.compileToMir(); props.close(); }}>
-      Build and show the resulting MIR, Rust's intermediate representation.
-    </ButtonMenuItem>
-    <ButtonMenuItem name="WASM" onClick={() => { props.compileToWasm(); props.close(); }}>
-      Build a WebAssembly module for web browsers, in the .WAT textual representation.
-      <p className="build-menu__aside">
-        Note: WASM currently requires using the Nightly channel, selecting this
-        option will switch to Nightly.
-      </p>
-    </ButtonMenuItem>
-  </MenuGroup >
-);
+const BuildMenu: React.SFC<BuildMenuProps> = props => {
+  // Show an aside on channels where WASM is not available yet
+  const wasmAside = props.channel !== Channel.Nightly ? (
+    <p className="build-menu__aside">
+      Note: WASM currently requires using the Nightly channel, selecting this
+      option will switch to Nightly.
+    </p>
+  ) : null;
+
+  return (
+    <MenuGroup title="What do you want to do?">
+      <ButtonMenuItem name="Build" onClick={() => { props.execute(); props.close(); }}>
+        No bells and whistles, regular build coming right&nbsp;up&nbsp;:D
+      </ButtonMenuItem>
+      <ButtonMenuItem name="ASM" onClick={() => { props.compileToAssembly(); props.close(); }}>
+        Build and show the resulting assembly code.
+      </ButtonMenuItem>
+      <ButtonMenuItem name="LLVM IR" onClick={() => { props.compileToLLVM(); props.close(); }}>
+        Build and show the resulting LLVM IR, LLVM's intermediate representation.
+      </ButtonMenuItem>
+      <ButtonMenuItem name="MIR" onClick={() => { props.compileToMir(); props.close(); }}>
+        Build and show the resulting MIR, Rust's intermediate representation.
+      </ButtonMenuItem>
+      <ButtonMenuItem name="WASM" onClick={() => { props.compileToWasm(); props.close(); }}>
+        Build a WebAssembly module for web browsers, in the .WAT textual representation.
+        {wasmAside}
```

Comment:
This can then become

```jsx
{ whateverYouCallTheProp  &&
<p className="build-menu__aside">
      Note: WASM currently requires using the Nightly channel, selecting this
      option will switch to Nightly.
    </p>
}
```

Or you could move the `p` to a top-level element for that nice one-liner:

```jsx
const WasmAside: React.SFC<{}> = props => (
  <p className="build-menu__aside">
     Note: WASM currently requires using the Nightly channel, selecting this
     option will switch to Nightly.
  </p>
);

{ whateverYouCallTheProp && <WasmAside /> }
```